evaluator/evaluator.py

```diff
@@ -270,31 +270,40 @@ def evaluate_vmaas(self, system_platform, cur, loop=None):
         new_cves = set()
         mitigated_cves_ids = set()
         unmitigated_cves_ids = set()
+        inc_cves_ids = set()
+        dec_cves_ids = set()
 
         for cve in reported_cves:
             if cve in system_cves_map:
                 unprocessed_cves.discard(cve)
                 if system_cves_map[cve]['when_mitigated']:
                     # it was mitigated, now its not
                     unmitigated_cves_ids.add(system_cves_map[cve]['cve_id'])
+                    if system_cves_map[cve]['rule_id'] is None:  # unmitigated only if there wasn't rule hit before
+                        inc_cves_ids.add(system_cves_map[cve]['cve_id'])
             else:
                 new_cves.add(cve)
 
+        rule_cves = 0
         for cve in unprocessed_cves:
             if not system_cves_map[cve]['when_mitigated']:
                 mitigated_cves_ids.add(system_cves_map[cve]['cve_id'])
+                if system_cves_map[cve]['rule_id'] is None:  # mitigated only in case there was not rule hit before
+                    dec_cves_ids.add(system_cves_map[cve]['cve_id'])
+                else:
+                    rule_cves += 1
+            elif system_cves_map[cve]['rule_id'] is not None:  # we have already mitigated CVE, but has rule accociated -> include in system count cache
```

Comment:
we have a CVE which was previously mitigated by VMaaS but it has rule associated so we have a case of system CVE coming only from rule engine